src/react-query/queryClient.ts

```diff
@@ -1,9 +1,22 @@
 import { MutationCache, QueryCache, QueryClient } from '@tanstack/react-query';
 import { toast } from 'sonner';
+import { AxiosError } from 'axios';
 
-function errorHandler(errorMsg: Error) {
+function errorHandler(rawErr: unknown) {
+	let errorMsg = 'We had some trouble!';
+	const axiosErr = rawErr as AxiosError<{ error?: string; message?: string; }>;
+	const otherErr = rawErr as { message?: string; };
+	if (axiosErr?.response?.data) {
+		if (axiosErr.response.data.error) {
+			errorMsg = axiosErr.response.data.error;
```

Comment:
This was the main culprit of the `[object Object]` errors showing up. We'd get `data`, which was an object, and it had something like: `{ "error": "failed because..." }`